### DISCUSSION TITLE
Paste text + link from buffer without `parent`

### DISCUSSION THREAD

**Discussion by HZ-labs** [ueberdosis/tiptap#1968] [General Questions] [answered]
https://github.com/ueberdosis/tiptap/blob/daa368a4cd725c7e03cbed2c21acd2e01e888c00/packages/core/src/pasteRules/markPasteRule.ts#L28

Why we cant use match result and generate a markup if we have match, but have not `parent`?

Perhaps should be like this `if (!parent || parent.type.allowsMarkType(type)) {...`

My case: https://www.loom.com/share/09b5ba8abcaa4cc1b764e6aa379b29dd

**Comment by philippkuehn** [comment, marked as answer]
yeah this is a bit broken. I’m currently rewriting inputrules and pasterules and keep that in mind ✌️
